src/main/java/com/openshift/jenkins/plugins/pipeline/model/IOpenShiftBuilder.java

```diff
@@ -128,35 +128,46 @@ default void waitOnBuild(IClient client, long startTime, String bldId, TaskListe
                 listener.getLogger().println("\nOpenShiftBuilder bld state:  " + bldState);
 
             if (needToFollow.compareAndSet(true, false)) {
-                final String container = pod.getContainers().iterator().next().getName();
-
-                stop = pod.accept(new CapabilityVisitor<IPodLogRetrievalAsync, IStoppable>() {
-                    @Override
-                    public IStoppable visit(IPodLogRetrievalAsync capability) {
-                        return capability.start(new IPodLogRetrievalAsync.IPodLogListener() {
-                            @Override
-                            public void onOpen() {
-                            }
-
-                            @Override
-                            public void onMessage(String message) {
-                                listener.getLogger().print(message);
-                            }
-
-                            @Override
-                            public void onClose(int code, String reason) {
-                            }
-
-                            @Override
-                            public void onFailure(IOException e) {
-                                // If follow fails, try to restart it on the next loop.
-                                needToFollow.compareAndSet(false, true);
-                            }
-                        }, new IPodLogRetrievalAsync.Options()
-                                .follow()
-                                .container(container));
-                    }
-                }, null);
+              List<IPod> pods = client.list(ResourceKind.POD, getNamespace(overrides));
+              for (IPod pod : pods) {
```

Comment:
Well, it is a little kludgy ... currently it is only set for source/docker/custom.  If "jenkinsfile" it is null.  I'll work with that, and add a TODO to update the restclient down the road.